armi/cases/tests/test_suiteBuilder.py

```diff
@@ -78,3 +96,78 @@ def test_addDegreeOfFreedom(self):
 
         with self.assertRaises(ValueError):
             builder.addDegreeOfFreedom([powerMod, morePowerMod])
+
+
+class TestFullFactorialSuiteBuilder(unittest.TestCase):
+    """Class to test FullFactorialSuiteBuilder."""
+
+    def test_buildSuite(self):
+        """Initialize a full factorial suite of cases.
+
+        .. test:: A generic mechanism to allow users to modify user inputs in cases.
+            :id: T_ARMI_CASE_MOD1
+            :tests: R_ARMI_CASE_MOD
+        """
+        builder = FullFactorialSuiteBuilder(case)
+        builder.addDegreeOfFreedom(
+            SettingModifier("settingName1", value) for value in (1, 2)
+        )
+        builder.addDegreeOfFreedom(
+            SettingModifier("settingName2", value) for value in (3, 4, 5)
+        )
+
+        assert builder.modifierSets[0][0].value == 1
+        assert builder.modifierSets[0][1].value == 3
+
+        assert builder.modifierSets[1][0].value == 2
+        assert builder.modifierSets[1][1].value == 3
+
+        assert builder.modifierSets[2][0].value == 1
+        assert builder.modifierSets[2][1].value == 4
+
+        assert builder.modifierSets[3][0].value == 2
+        assert builder.modifierSets[3][1].value == 4
+
+        assert builder.modifierSets[4][0].value == 1
+        assert builder.modifierSets[4][1].value == 5
+
+        assert builder.modifierSets[5][0].value == 2
+        assert builder.modifierSets[5][1].value == 5
+
+        assert len(builder.modifierSets) == 6
```

Comment:
Can you change these to use `self.assertEquals()`?

All of our unit tests use that logic; it reports better through PyTest.

Thanks!

Reply:
@john-science Converted asserts to self.asserts.